Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -11575,42 +11576,38 @@ void WebPageProxy::setOverlayScrollbarStyle(std::optional<WebCore::ScrollbarOver
         legacyMainFrameProcess().send(Messages::WebPage::SetScrollbarOverlayStyle(scrollbarStyleForMessage), internals().webPageID);
 }
 
-std::optional<Vector<uint8_t>> WebPageProxy::getWebCryptoMasterKey()
+void WebPageProxy::getWebCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
 {
-    if (auto keyData = m_websiteDataStore->client().webCryptoMasterKey())
-        return keyData;
-    if (auto keyData = m_navigationClient->webCryptoMasterKey(*this))
-        return Vector(keyData->span());
-    return std::nullopt;
+    m_navigationClient->legacyWebCryptoMasterKey(*this, [completionHandler = WTFMove(completionHandler), this](std::optional<Vector<uint8_t>>&& key) mutable {
```

Comment:
We need protectedThis = Ref { *this } or else use of this could be a UAF.  Or a weakThis with a null check instead of capturing this.